### PR TITLE
docs(vehicle component design): add vehicle component design doc

### PR DESCRIPTION
**Pull request by LalithVipulananthan**
## Description

Adding the vehicle interface design doc to Autoware documentation to make it easier to collate feedback from other Autoware developers.

I am still in the process of trying to find examples of performance requirements by ODD, including how those requirements were calculated so that other developers can adapt those calculations for their ODD.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/design/autoware-architecture/vehicle/index.md [hunk 1]
```diff
@@ -1,5 +1,117 @@
...
+
+<!-- insert vehicle component architecture diagram here
+![alt_text](images/image1.png "image_tooltip")
+-->
+
+<!-- insert vehicle component overview diagram here
+![alt_text](images/image1.png "image_tooltip")
+-->
+
+## 3. Features
```
Comment (mitsudome-r) [inline docs/design/autoware-architecture/vehicle/index.md:37]: Currently `Features` include:
- Diagnostics
- Safety features
- Optional controls

we probably should include basic function as well:
- Converting Autoware control commands to vehicle specific command
- Converting vehicle specific status information (velocity, steering) to Autoware status message

Although we already explain them in the "Requirements" section, I think we should mention it again.

File: docs/design/autoware-architecture/vehicle/index.md [hunk 2]
```diff
@@ -1,5 +1,117 @@
-# Vehicle component design
+# Vehicle interface design
```
Comment (LalithVipulananthan) [inline docs/design/autoware-architecture/vehicle/index.md]:
```suggestion
# Vehicle Interface design
```

File: docs/design/autoware-architecture/vehicle/index.md [hunk 3]
```diff
@@ -1,5 +1,117 @@
...
+The Vehicle Interface component consists of the following components:
+
+- A Raw Vehicle Command Converter component that will pass through vehicle commands from the Control component if velocity/acceleration control is supported by the drive-by-wire system. Otherwise, the Control commands will be modified according to the control method (eg: converting a target acceleration from the Control component to a vehicle specific accel/brake pedal value through the use of an acceleration map)
+- A Vehicle Interface component (vehicle specific) that acts as an interface between Autoware and a vehicle to communicate control signals and to obtain information about the vehicle (steer output, tyre angle etc)
+
+Each component contains static nodes of Autoware, while each module can be dynamically loaded and unloaded (corresponding to C++ classes). The mechanism of the Vehicle Interface component is depicted by the following figures:
+
+<!-- insert vehicle component architecture diagram here
+![alt_text](images/image1.png "image_tooltip")
+-->
```
Comment (LalithVipulananthan) [inline docs/design/autoware-architecture/vehicle/index.md:31]:
```suggestion
![Vehicle Interface overview](..image/vehicle_interface_architecture.png)
```

File: docs/design/autoware-architecture/vehicle/index.md [hunk 4]
```diff
@@ -1,5 +1,117 @@
...
+
+<!-- insert vehicle component overview diagram here
+![alt_text](images/image1.png "image_tooltip")
+-->
```
Comment (LalithVipulananthan) [inline docs/design/autoware-architecture/vehicle/index.md:35]:
```suggestion
![Vehicle Interface overview](..image/vehicle_interface_overview.png)
```

File: docs/design/autoware-architecture/vehicle/index.md [hunk 5]
```diff
@@ -1,5 +1,113 @@
...
+Each component contains static nodes of Autoware, while each module can be dynamically loaded and unloaded (corresponding to C++ classes). The mechanism of the Vehicle Interface component is depicted by the following figures:
+
+![Vehicle Interface overview](..image/vehicle_interface_architecture.png)
+
+![Vehicle Interface overview](..image/vehicle_interface_overview.png)
+
+## 3. Features
+
+The Vehicle Interface component can provide the following features in functionality and capability:
+
```
Comment (LalithVipulananthan) [inline docs/design/autoware-architecture/vehicle/index.md:36]:
```suggestion

- Basic functions
  - Converting Autoware control commands to vehicle specific command
  - Converting vehicle specific status information (velocity, steering) to Autoware status message
  
```

File: docs/design/autoware-architecture/vehicle/index.md [hunk 6]
```diff
@@ -1,5 +1,117 @@
...
+- Diagnostics
+  - List available features
+  - Provide a warning if the Control component tries to use a feature that is not available in the DBW component
+
+Additional functionality and capability features may be added, depending on the vehicle hardware. Some example features are listed below:
+
+- Safety features
+  - Disengage autonomous driving via manual intervention.
+    - This can be done through the use of an emergency disengage button, or by a safety driver manually turning the steering wheel or pressing the brake
+- Optional controls
```
[the remark this answers is not in the document]
Comment (LalithVipulananthan) [reply docs/design/autoware-architecture/vehicle/index.md:46]: I thought that this point was covered by the diagnostics feature "Provide a warning if the Control component tries to use a feature that is not available in the Vehicle Interface component". Is this not sufficient?

File: docs/design/autoware-architecture/vehicle/index.md [hunk 7]
```diff
@@ -1,5 +1,118 @@
...
+The Vehicle Interface component consists of the following components:
+
+- A Raw Vehicle Command Converter component that will pass through vehicle commands from the Control component if velocity/acceleration control is supported by the drive-by-wire system. Otherwise, the Control commands will be modified according to the control method (eg: converting a target acceleration from the Control component to a vehicle specific accel/brake pedal value through the use of an acceleration map)
+- A Vehicle Interface component (vehicle specific) that acts as an interface between Autoware and a vehicle to communicate control signals and to obtain information about the vehicle (steer output, tyre angle etc)
+
+Each component contains static nodes of Autoware, while each module can be dynamically loaded and unloaded (corresponding to C++ classes). The mechanism of the Vehicle Interface component is depicted by the following figures:
+
+![Vehicle Interface overview](..image/vehicle_interface_architecture.png)
+
+![Vehicle Interface overview](..image/vehicle_interface_overview.png)
```
Comment (LalithVipulananthan) [inline docs/design/autoware-architecture/vehicle/index.md:31]:
```suggestion
![Vehicle Interface overview](./image/vehicle_interface_architecture.png)

![Vehicle Interface overview](./image/vehicle_interface_overview.png)
```

File: docs/design/autoware-architecture/vehicle/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mitsudome-r) [reply docs/design/autoware-architecture/vehicle/index.md:51]: Thanks, I overlooked that. I think it's okay as is then.

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**